sdk/src/main/scala/com/horizen/account/mempool/MempoolMap.scala

```diff
@@ -175,7 +170,197 @@ class MempoolMap(stateReader: AccountStateReader) extends ScorexLogging {
   }
 
   def canPayHigherFee(newTx: SidechainTypes#SCAT, oldTx: SidechainTypes#SCAT): Boolean = {
-     (newTx.getMaxFeePerGas.compareTo(oldTx.getMaxFeePerGas) > 0) &&
-      (newTx.getMaxPriorityFeePerGas.compareTo(oldTx.getMaxPriorityFeePerGas) > 0)
+    (newTx.getMaxFeePerGas.compareTo(oldTx.getMaxFeePerGas) > 0) &&
+    (newTx.getMaxPriorityFeePerGas.compareTo(oldTx.getMaxPriorityFeePerGas) > 0)
+  }
+
+  def updateMemPool(rejectedBlocks: Seq[AccountBlock], appliedBlocks: Seq[AccountBlock]): Unit = {
```

Comment:
As for me, comments for every logical block here and sub-methods are very useful.
A lot of modifications are done considering some assumptions.

Reply:
Done